src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -60,32 +62,35 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
     }
 
     renderButtons(paypalCheckoutInstance: PaypalClientInstance) {
-        const allowedSources = [];
-        const disallowedSources = [];
         const { paypalOptions, paymentMethod, container } = this._renderButtonsData as RenderButtonsData;
         const { paypal } = this._window;
 
         if (paypal) {
-            if (paypalOptions.allowCredit) {
-                allowedSources.push(paypal.FUNDING.CREDIT);
-            } else {
-                disallowedSources.push(paypal.FUNDING.CREDIT);
+            const FUNDING_SOURCES = [];
+            for (const fundingKey in paypal.FUNDING) {
```

Comment:
We need to pass CREDIT funding source if it has been enabled in options, e.g. like here:
```
            if (paypalOptions.allowCredit) {
                allowedSources.push(paypal.FUNDING.CREDIT);
            } else {
                disallowedSources.push(paypal.FUNDING.CREDIT);
```
Have you checked following solution to avoid for loop?
https://developer.paypal.com/docs/checkout/integration-features/standalone-buttons/#render-a-set-of-buttons